tests/unit/commands/test_remote.py

```diff
@@ -769,6 +769,124 @@ def test_platform_core22_error(
     assert "Use '--build-for' instead." in err
 
 
+@pytest.mark.parametrize(
+    ("base", "build_info", "error_messages"),
+    [
+        pytest.param(
+            "core22",
+            {
+                "architectures": [
+                    {"build-on": "amd64", "build-for": "amd64"},
+                    {"build-on": ["amd64", "riscv64"], "build-for": "riscv64"},
+                ],
+            },
+            ["Building on 'amd64' will create snaps for 'amd64' and 'riscv64'."],
+            id="core22-simple",
+        ),
+        pytest.param(
+            "core22",
+            {
+                "architectures": [
+                    {"build-on": "amd64", "build-for": "amd64"},
+                    {"build-on": ["amd64", "riscv64"], "build-for": "riscv64"},
+                    {"build-on": "s390x", "build-for": "s390x"},
+                    {"build-on": "s390x", "build-for": "ppc64el"},
+                ],
+            },
+            [
+                "Building on 'amd64' will create snaps for 'amd64' and 'riscv64'.",
+                "Building on 's390x' will create snaps for 'ppc64el' and 's390x'.",
+            ],
+            id="core22-complex",
+        ),
+        pytest.param(
+            "core24",
+            {
+                "platforms": {
+                    "platform1": {"build-on": "amd64", "build-for": "amd64"},
+                    "platform2": {
+                        "build-on": ["amd64", "riscv64"],
+                        "build-for": "riscv64",
+                    },
+                },
+            },
+            ["Building on 'amd64' will create snaps for 'amd64' and 'riscv64'."],
+            id="core24-simple",
+        ),
+        pytest.param(
+            "core24",
+            {
+                "platforms": {
+                    "platform1": {"build-on": "amd64", "build-for": "riscv64"},
+                    "identical-platform": {"build-on": "amd64", "build-for": "riscv64"},
+                },
+            },
+            # this will show 'riscv64' twice because the platform name is different
+            ["Building on 'amd64' will create snaps for 'riscv64' and 'riscv64'."],
```

Comment:
This could be improved but I'm leaving this as-is because:
- it tells the user everything they need, even if it may look odd
- this is a 3-point complexity ticket
- this is an uncommon scenario
- our time would be better spent creating a `__str__()` method for the `BuildInfo` class in craft-application